front/app/modules/commercial/insights/admin/containers/Insights/Edit/Categories/Categories.test.tsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import { render, screen, fireEvent, act } from 'utils/testUtils/rtl';
+import * as service from 'modules/commercial/insights/services/insightsCategories';
+import clHistory from 'utils/cl-router/history';
+
+import Categories from './';
+
+let mockData = [
+  {
+    id: '1aa8a788-3aee-4ada-a581-6d934e49784b',
+    type: 'category',
+    attributes: {
+      name: 'Test',
+      inputs_count: 3,
+    },
+  },
+  {
+    id: '4b429681-1744-456f-8550-e89a2c2c74b2',
+    type: 'category',
+    attributes: {
+      name: 'Test 2',
+      inputs_count: 5,
+    },
+  },
+];
+
+const viewId = '1';
+
+jest.mock('utils/cl-intl');
+
+jest.mock('modules/commercial/insights/services/insightsCategories', () => ({
+  addInsightsCategory: jest.fn(),
+  deleteInsightsCategories: jest.fn(),
+  deleteInsightsCategory: jest.fn(),
+}));
+
+jest.mock('modules/commercial/insights/hooks/useInsightsCategories', () => {
+  return jest.fn(() => mockData);
+});
+
+const allInputsCount = 10;
+const uncategorizedInputCount = 5;
+
+jest.mock('modules/commercial/insights/hooks/useInsightsInputsCount', () => {
+  return jest.fn((_viewId, queryParameters) => {
+    return queryParameters === undefined
+      ? { count: allInputsCount }
+      : { count: uncategorizedInputCount };
+  });
+});
+
+jest.mock('hooks/useLocale', () => jest.fn(() => 'en'));
+
+const mockLocationData = { pathname: '', query: {} };
+
+jest.mock('react-router', () => {
+  return {
+    withRouter: (Component) => {
+      return (props) => {
+        return (
+          <Component
+            {...props}
+            params={{ viewId }}
+            location={mockLocationData}
+          />
+        );
+      };
+    },
+    Link: () => 'Link',
+  };
+});
+
+window.confirm = jest.fn(() => true);
```

Comment:
Would be nice to add some kind of global mock for that, maybe in testUtils/rtl ?

Reply:
Good idea! 👍